src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+        successCallback: Function,
+        errorCallback: Function
+    ): void => {
+        const uuid = uniqueId('comment_');
+        const commentData = {
+            id: uuid,
+            tagged_message: text,
+            type: 'comment'
+        };
+
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.addPendingItem(this.id, currentUser, commentData);
+
+        const message = {};
+        if (hasMention) {
+            message.taggedMessage = text;
+        } else {
+            message.message = text;
+        }
+
+        this.commentsAPI = new CommentsAPI(this.options);
+
+        this.commentsAPI.createComment({
+            file,
+            ...message,
+            successCallback: (comment: Comment) => {
+                this.createCommentSuccessCallback(comment, uuid, successCallback);
+            },
+            errorCallback: (e: $AxiosXHR<any>) => {
+                this.createCommentErrorCallback(e, uuid, errorCallback);
+            }
+        });
+    };
+
+    /**
+     * Adds the current version from the file object, which may be a restore
+     *
+     * @param {FileVersions} versions - API returned file versions for this file
+     * @return {FileVersions} modified versions array including the current/restored version
+     */
+    addCurrentVersion(versions: ?FileVersions, file: BoxItem): ?FileVersions {
+        const { restored_from, modified_at, file_version } = file;
+
+        if (!file_version || !versions) {
+            return versions;
+        }
+
+        const { modified_by, version_number } = file;
+        let currentVersion = file_version;
+        let action = VERSION_UPLOAD_ACTION;
+        let versionNumber = version_number;
+
+        if (restored_from) {
+            const { id: restoredFromId } = restored_from;
+            const restoredVersion = versions.entries.find((version: BoxItemVersion) => version.id === restoredFromId);
+            if (restoredVersion) {
+                versionNumber = restoredVersion.version_number;
+                action = VERSION_RESTORE_ACTION;
+                currentVersion = {
+                    ...restoredVersion,
+                    ...currentVersion
+                };
+            }
+        }
+
+        const currentFileVersion: BoxItemVersion = {
+            ...currentVersion,
+            action,
+            modified_by,
+            created_at: modified_at,
+            version_number: versionNumber
+        };
+        return {
+            total_count: versions.total_count + 1,
+            entries: [...versions.entries, currentFileVersion]
+        };
+    }
+
+    /**
+     * Destroys all the task assignment API's
+     *
+     * @return {void}
+     */
+    destroyTaskAssignments() {
+        if (Array.isArray(this.taskAssignmentsAPI)) {
+            this.taskAssignmentsAPI.forEach((api) => api.destroy());
+            this.taskAssignmentsAPI = [];
```

Comment:
What's our standard on emptying arrays? Creating a new array, or reusing the one we have via `array.length =  0`?

Reply:
I have heard of this approach but have not dug into the details of what happens when you set the length to 0. Does it remove all the references?